src/data/fintype/basic.lean

```diff
@@ -2010,6 +2010,13 @@ infinite.of_surjective (coe : list α → multiset α) (surjective_quot_mk _)
 instance [infinite α] : infinite (option α) :=
 infinite.of_injective some (option.some_injective α)
 
+-- really we only need one infinite and the others inhabited
```

Comment:
I wonder if perhaps you could prove this as a lemma, and use it to build the less general instance.